src/lib/operators/table_scan.cpp

```diff
@@ -0,0 +1,148 @@
+#include "table_scan.hpp"
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <storage/reference_segment.hpp>
+#include <storage/dictionary_segment.hpp>
+#include <storage/table.hpp>
+#include <storage/value_segment.hpp>
+#include "resolve_type.hpp"
+
+namespace opossum {
+
+TableScan::TableScan(const std::shared_ptr<const AbstractOperator>& in, const ColumnID column_id,
+                     const ScanType scan_type, const AllTypeVariant search_value)
+    : _in_operator(in), _column_id(column_id), _scan_type(scan_type), _search_value(search_value) {}
+
+ColumnID TableScan::column_id() const { return _column_id; }
+
+ScanType TableScan::scan_type() const { return _scan_type; }
+
+const AllTypeVariant& TableScan::search_value() const { return _search_value; }
+
+template <typename T>
+std::function<bool(T, T)> TableScan::_get_comparator_method() {
+  std::function<bool(T, T)> _comp;
+
+  switch (_scan_type) {
+    case ScanType::OpGreaterThanEquals:
+      _comp = [&](const T search_value, const T comparison_value) { return search_value <= comparison_value; };
+      break;
+    case ScanType::OpGreaterThan:
+      _comp = [&](const T search_value, const T comparison_value) { return search_value < comparison_value; };
+      break;
+    case ScanType::OpLessThanEquals:
+      _comp = [&](const T search_value, const T comparison_value) { return search_value >= comparison_value; };
+      break;
+    case ScanType::OpLessThan:
+      _comp = [&](const T search_value, const T comparison_value) { return search_value > comparison_value; };
+      break;
+    case ScanType::OpNotEquals:
+      _comp = [&](const T search_value, const T comparison_value) { return search_value != comparison_value; };
+      break;
+    case ScanType::OpEquals:
+      _comp = [&](const T search_value, const T comparison_value) { return search_value == comparison_value; };
+      break;
+    default:
+      throw "Operator not implemented";
+  }
+
+  return _comp;
+}
+
+std::shared_ptr<const Table> TableScan::_on_execute() {
+  auto input_table = _in_operator->get_output();
+
+  // get characteristics for output table
+  auto col_type = input_table->column_type(_column_id);
+  auto column_count = input_table->column_count();
+
+  // create empty output table
+  auto output_table = std::make_shared<Table>(input_table->target_chunk_size());
+  auto pos_list = std::make_shared<PosList>();
+
+  // check if input table contains reference segments
+  auto& probe_chunk = input_table->get_chunk(ChunkID{0});
+  const auto& probe_segment = probe_chunk.get_segment(ColumnID{0});
+  auto probe_ref_segment = std::dynamic_pointer_cast<ReferenceSegment>(probe_segment);
+  if (probe_ref_segment) {
```

Comment:
I would do the commenting similar to how it is below, even though it's kinda obvious, but maybe it helps to make the code more readable.
```suggestion
  if (probe_ref_segment) {
  //ref segment
```